src/components/app-scope.js

```diff
@@ -14,41 +46,200 @@ class AppScope extends React.Component {
     }
     return dict
   }
+  convertAlreadySelectedDictionary = options => {
+    let dict = {}
+    for (let i = 0; i < options.length; i++) {
+      if(!dict[options[i]]){
+        dict[options[i]]=true;
+      }
+    }
+    return dict
+  }
+  convertOptionsListDictionary = options => {
+    let dict = {}
+    for (let i = 0; i < options.length; i++) {
+      if (!dict[options[i].value.split('.')[0]]) {
+        dict[options[i].value.split('.')[0]] = []
+      }
+      dict[options[i].value.split('.')[0]].push(options[i])
+    }
+    return dict
+  }
+  handleCheckboxChange = (e, { checked }) => {
+    const prevScope = this.state.scope
+    prevScope[e.target.id] = checked
+    this.setState({
+      scope: prevScope
+    })
+  }
+  handleClick = () => {
+    const {
+      scope
+    } = this.state
+    if (this.props.activeApp.inEditMode === 'none') {
+      if (this.state.editMode) {
+        const dataPoints = Object.keys(scope).filter(sco => {
+          if (scope[sco]) {
+            return sco
+          }
+        })
+        this.props.ChangeActiveApp(
+          this.props.activeApp.data.id,
+          'dataPoints',
+          {
+            data_points: dataPoints
+          },
+          this.successCallback,
+          this.errCallback
+        )
+      } else {
+        this.setState({
+          editMode: !this.state.editMode
+        })
+      }
+    }
+  }
+  successCallback = res => {
+    this.setState({
+      error: false,
+      message: '',
+      editMode: !this.state.editMode
+    })
+  }
+  errCallback = err => {
+    this.setState({
+      error: true,
+      message: err.response.data
+    })
+  }
   render () {
+    const { optionsList } = this.props
     const { activeApp } = this.props
-    const { data } = activeApp
+    const { editMode, error, message } = this.state
+    const { data, inEditMode } = activeApp
     return (
       <React.Fragment>
-        <Segment attached='top'>
-          <Header as='h4'>Scope</Header>
+        <Segment attached='top' styleName='app-scope-segment'>
+          <Header as='h4' styleName='app-scope-header'>Scope</Header>
+          {editMode || error ? (
```

Comment:
Why are we showing same icon for `editMode` and `error`?

Reply:
This is the save icon.
If error is present, it will be shown as a written text in red inside the scope segment. The save icon should not be changed due to an error.